api/integrations/github/views.py

```diff
@@ -204,3 +223,27 @@ def fetch_repositories(request, organisation_pk: int):
         return Response(data)
     except requests.RequestException as e:
         return JsonResponse({"error": str(e)}, status=500)
+
+
+@api_view(["POST"])
+@permission_classes([AllowAny])
+def github_webhook(request) -> JsonResponse:
+    # Validate payload signature
+    secret = settings.GITHUB_WEBHOOK_SECRET
+    signature = request.headers.get("X-Hub-Signature")
+    github_event = request.headers.get("x-github-event")
+    payload = request.body
+    if github_webhook_payload_is_valid(
+        payload_body=payload, secret_token=secret, signature_header=signature
+    ):
+        data = json.loads(payload.decode("utf-8"))
+        # handle GitHub Webhook "installation" event with action type "deleted"
+        if github_event == "installation" and data["action"] == "deleted":
+            GithubConfiguration.objects.filter(
+                installation_id=data["installation"]["id"]
+            ).delete()
+            return JsonResponse({"detail": "Event processed"}, status=200)
```

Comment:
Is there a reason why we are not using `Response` here?

Reply:
Not Really I will change this and other views to use Response

Reply:
Not Really I will change this and other views to use Response

Reply:
Done